internal/tsoptions/tsconfigparsing.go

```diff
@@ -520,7 +528,7 @@ func convertOptionsFromJson[O optionParser](optionsNameMap map[string]*CommandLi
 		opt, ok := optionsNameMap[key]
 		if !ok {
 			// !!! TODO?: support suggestion
-			errors = append(errors, ast.NewCompilerDiagnostic(diagnostics.Unknown_compiler_option_0, key))
+			errors = append(errors, createUnknownOptionError(key, result.UnknownOptionDiagnostic(), "", nil, nil, nil))
```

Comment:
isnt this just `result.UnknownOptionDiagnostic()` after adding that method to `optionParser`

Reply:
`result.UnknownOptionDiagnostic()` only returns the diagnostic message. We need the call to create the full error (and format if needed)